staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_test.go

```diff
@@ -364,96 +304,6 @@ func TestNamespaceScopedWatch(t *testing.T) {
 	storagetesting.RunTestNamespaceScopedWatch(ctx, t, cacher)
 }
 
-// TODO(wojtek-t): We should extend the generic RunTestWatch test to cover the
-// scenarios that are not yet covered by it and get rid of this test.
-func TestWatchDeprecated(t *testing.T) {
-	server, etcdStorage := newEtcdTestStorage(t, etcd3testing.PathPrefix(), true)
-	defer server.Terminate(t)
-	fakeClock := testingclock.NewFakeClock(time.Now())
-	cacher, _, err := newTestCacherWithClock(etcdStorage, fakeClock)
-	if err != nil {
-		t.Fatalf("Couldn't create cacher: %v", err)
-	}
-	defer cacher.Stop()
-
-	podFoo := makeTestPodWithName("foo")
-	podBar := makeTestPodWithName("bar")
-
-	podFooPrime := makeTestPodWithName("foo")
-	podFooPrime.Spec.NodeName = "fakeNode"
-
-	podFooBis := makeTestPodWithName("foo")
-	podFooBis.Spec.NodeName = "anotherFakeNode"
-
-	podFooNS2 := makeTestPodWithName("foo")
-	podFooNS2.Namespace += "2"
-
-	// initialVersion is used to initate the watcher at the beginning of the world,
-	// which is not defined precisely in etcd.
-	initialVersion, err := cacher.LastSyncResourceVersion()
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	startVersion := strconv.Itoa(int(initialVersion))
-
-	// Set up Watch for object "podFoo".
-	watcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", storage.ListOptions{ResourceVersion: startVersion, Predicate: storage.Everything})
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer watcher.Stop()
-
-	// Create in another namespace first to make sure events from other namespaces don't get delivered
-	updatePod(t, etcdStorage, podFooNS2, nil)
-
-	fooCreated := updatePod(t, etcdStorage, podFoo, nil)
-	_ = updatePod(t, etcdStorage, podBar, nil)
-	fooUpdated := updatePod(t, etcdStorage, podFooPrime, fooCreated)
-
-	verifyWatchEvent(t, watcher, watch.Added, podFoo)
-	verifyWatchEvent(t, watcher, watch.Modified, podFooPrime)
-
-	initialWatcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", storage.ListOptions{ResourceVersion: fooCreated.ResourceVersion, Predicate: storage.Everything})
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer initialWatcher.Stop()
-
-	verifyWatchEvent(t, initialWatcher, watch.Modified, podFooPrime)
-
-	// Now test watch from "now".
-	nowWatcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", storage.ListOptions{ResourceVersion: "0", Predicate: storage.Everything})
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer nowWatcher.Stop()
-
-	verifyWatchEvent(t, nowWatcher, watch.Added, podFooPrime)
-
-	_ = updatePod(t, etcdStorage, podFooBis, fooUpdated)
-
-	verifyWatchEvent(t, nowWatcher, watch.Modified, podFooBis)
-
-	// Add watchCacheDefaultCapacity events to make current watch cache full.
```

Comment:
FWIW - all cases above are already being tested by the existing TestWatch test.
The functionality of expiriing cache was added to TestWatchFromZero below.

So we're now safe to remove it.